src/api/cvc4cpp.h

```diff
@@ -0,0 +1,4556 @@
+   */
+  ITE,
+
+  /* UF -------------------------------------------------------------------- */
+
+  /* Application of an uninterpreted function.
+   * Parameters: n > 1
+   *   -[1]: the function Term
+   *   -[2]..[n]: function argument instantiation Terms
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  APPLY_UF,
+  /**
+   * Boolean term variable
+   * ???
+   */
+  BOOLEAN_TERM_VARIABLE,
+  /**
+   * Cardinality constraint on sort S.
+   * Parameters: 2
+   *   -[1]: a Term of Sort S
+   *   -[2]: a positive integer constant that bounds the cardinality of sort S
+   * Create with:
+   *   ???
+   */
+  CARDINALITY_CONSTRAINT,
+  /**
+   * Combined cardinality constraint.
+   * Parameters: 1
+   *   -[1]: a positive integer constant that bounds the sum of the
+   *         cardinalities of all sorts in the signature
+   * Create with:
+   *   ???
+   */
+  COMBINED_CARDINALITY_CONSTRAINT,
+  /**
+   * Partial uninterpreted function application.
+   * Parameters: n  ??? > 1 ???
+   *   -[1]..[n]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  PARTIAL_APPLY_UF,
+#if 0
+  /* cardinality value of sort S:
+   * first parameter is (any) term of sort S */
+   CARDINALITY_VALUE,
+#endif
+  /**
+   * Higher-order (partial) function application
+   * Parameters: n  ???  n > 1 ???
+   *   -[1]..[n]: ???
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  HO_APPLY,
+
+  /* Arithmetic ------------------------------------------------------------ */
+
+  /**
+   * Arithmetic addition.
+   * Parameters: n > 1
+   *   -[1]..[n]: Terms of Sort Integer, Real (sorts must match).
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  PLUS,
+  /**
+   * Arithmetic multiplication.
+   * Parameters: n > 1
+   *   -[1]..[n]: Terms of Sort Integer, Real (sorts must match).
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  MULT,
+  /* Synonym for MULT.
+   * Parameters: n > 1
+   *   -[1]..[n]: Terms of Sort Integer, Real (sorts must match).
+   * Create with:
+   *   mkTerm(Kind kind, Term child1, Term child2)
+   *   mkTerm(Kind kind, Term child1, Term child2, Term child3)
+   *   mkTerm(Kind kind, const std::vector<Term>& children)
+   */
+  NONLINEAR_MULT,
```

Comment:
This should also be internal only, right?  @ajreynol ?

Reply:
Yes, this is internal.